Woo/order detail shipment fix

anitaa1990 opened the pull request:
Fixes #2801 and #2800.  

#### Issue 2800
Show option to add tracking in order details if plugin is available. Fixed in 666147c.

##### To test
- Enable the Shipment Tracking plugin on your test site.
- Click on an order from the app.
- Notice the "Add tracking" section in the Order Detail screen.
- Disable the Shipment Tracking plugin on your test site.
- Click on an order from the app.
- Notice that the "Add tracking" section is no longer displayed in the Order Detail screen.

#### Issue 2801
After changing an order's details (e.g. adding an order note or issuing a refund) the Products section disappears from the order details screen. If I try to pull to refresh the screen at that point, I see loading indicators but it never stops loading.

Fixed in 464d2ce

##### To test
- Go to the Orders tab and select an order.
- Confirm the order details screen includes all the expected information.
- Go to the order notes section and add a note.
- Scroll back up and notice the Products section is now visible.
- Pull to refresh and notice the screen loads as expected.

Thank you for flagging these issues @rachelmcr! I have added you as a reviewer since you caught the initial issue. 

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

anitaa1990 committed:
Show option to add tracking in order details if ST plugin is available

anitaa1990 committed:
Fixed infinite skeleton loading bug in order detail when order is refreshed

anitaa1990 committed:
Fixed failing unit test

AmandaRiu reviewed (changes requested):
@anitaa1990  Tested through both scenarios and the fix to display tracking information is working (though you have to leave an order and come back into the order to see changes to the plugin status -- PTR doesn't work). 

The second part of this PR does not appear to be fixed. If I add a note to an order the products section will disappear and won't return. PTR does not bring the products card back.

anitaa1990 committed:
Display product list by default since shipping labels is not yet live

anitaa1990 committed:
Added logic to hide shipment tracking section if plugin is not available when PTR is initiated

anitaa1990 committed:
Fixed failing unit test

anitaa1990 commented:
Thanks for the review @AmandaRiu! 

> The second part of this PR does not appear to be fixed. If I add a note to an order the products section will disappear and won't return. PTR does not bring the products card back.

I have removed the logic to hide the product list since it is not needed when we release the shipping labels project and there are some changes related to that I've been working on in a separate feature branch.

> he fix to display tracking information is working (though you have to leave an order and come back into the order to see changes to the plugin status -- PTR doesn't work).

Nice catch! This was because if the shipment tracking section was displayed and we deactivate the plugin and initiate PTR, there was no option to hide the tracking section. I've fixed that in ba8a9f0.

Ready for another round 👍

AmandaRiu reviewed (approved):
Changes look good. Thanks @anitaa1990 ! :shipit:
